### DISCUSSION TITLE
Indicate number of Power Platform Requests consumed by low-code plugin

### DISCUSSION THREAD

**Discussion by filcole** [microsoft/Power-Fx#2383] [Ideas]
It would be great to see the number of Power Platform requests that in a low code plugin will consume.

## At design time

For instant plugins I'm guessing that it would be the dataverse event for the plugin, plus any additional power platform events (Collect, or Patch or Connectors)

For low-code plugins I'm guessing it would be the same as above, would there be any difference?

If the PowerFx consist of conditional statements (if/switch) then might it be possible to determine the minimum and maximum number of requests possible via static analysis?

## At run time

Could the actual number of PP requests be included alongside the logging of the low-code plugin?  (as seen in the new dataverse accelerator, or perhaps the tranditional plugin trace log?)

**Comment by MikeStall** [comment]
Similar to the monitor tool in PA?
https://learn.microsoft.com/en-us/power-apps/maker/monitor-overview

**Comment by filcole** [comment]
Yes, similar, but it's unclear exactly what in PA monitor counts as a pp request. 

Having at design time would be great too.

**Comment by robwood-TTECDigital** [comment]
I vote for the runtime option because the design time option looks hard - wouldn't it just give 0..N at lot of times because it depends on how many rows are being returned by queries?

The runtime option [Edit: to plugin trace log] would be useful for plugins in general and not just LCPs - although it's easier to work out ahead of time for trad plugins.

If it could also log what those requests were with the basic details, that would be useful too because that's the next question once you see a high number of requests - what are they?
